core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -192,6 +192,11 @@ protected void cleanUncommitted(Set<ManifestFile> committed) {
     }
   }
 
+  @Override
+  protected boolean canSkipCleanupAfterCommitSuccess() {
+    return commitMetrics().attempts().value() == 1;
+  }
```

Comment:
I think I'd rename this to `cleanupAfterCommit` and invert the logic to `commitMetrics().attempts().value() > 1`. Makes it more readable imo.